src/symmetric_heap_c.c

```diff
@@ -396,3 +399,27 @@ void SHMEM_FUNCTION_ATTRIBUTES * shmemalign(size_t alignment, size_t size)
 {
     return shmem_align(alignment, size);
 }
+
+
+void SHMEM_FUNCTION_ATTRIBUTES *
+shmemx_malloc_with_hints(size_t size, long hints)
+{
+    void *ret = NULL;
+
+    SHMEM_ERR_CHECK_INITIALIZED();
+
+    if (size == 0) return ret;
+
+    // Check for valid hints
+    if(hints >= SHMEMX_MALLOC_MAX_HINTS || hints < 0) {
```

Comment:
`hints == SHMEMX_MALLOC_MAX_HINTS` is ok, right? Does the unit test check the `SHMEMX_MALLOC_ATOMICS_REMOTE | SHMEMX_MALLOC_SIGNAL_REMOTE` case?

Reply:
yeah the unit tests does check for the combination so SHMEMX_MALLOC_MAX_HINTS is also a valid hint.

Reply:
```suggestion
    if(hints > SHMEMX_MALLOC_MAX_HINTS || hints < 0) {
```
It looks like `SHMEMX_MALLOC_MAX_HINTS == SHMEMX_MALLOC_ATOMICS_REMOTE | SHMEMX_MALLOC_SIGNAL_REMOTE`, and this is a valid hint. Am I reading this right?

Reply:
Yeah that's right! I'll fix this.